### DISCUSSION TITLE
DPI Switch and prevent DPI warning

### DISCUSSION THREAD

**Discussion by Simba98** [Xpra-org/xpra#3993] [Q&A] [answered]
![image](https://github.com/Xpra-org/xpra/assets/7941967/a7e31f96-281e-4bcb-9f27-6beb9642ad87)
Here is the DPI warning, And I have two monitors, one is 100% DPI and the other is 150% DPI.
I think I don't need DPI scale, so I use the following command
`&"C:\Program Files\Xpra\Xpra.exe" start ssh/HOST/PORT --desktop-scaling=off --dpi=48`
But still have the above warning message?

What should I do to prevent DPI warning? Is it related to the server side? Server is Ubuntu 20.04.

Thank you in advance.

**Comment by totaam** [comment]
What distribution, version, etc? Server log?
My guess is that you're running Debian or Ubuntu, and these distros have DPI issues.

**Reply by Simba98** [reply]
Sorry for the incomplete information, I was guessing this is a client-only issue. 

Yes, your guess is 100% correct, I am running on Ubuntu 20.04. The client is Windows 11, and the server is Ubuntu 20.04.
And Xpra client is 4.4.6.
Do you recommend using a docker to run xpra on a different "OS"?

**Reply by totaam** [reply, marked as answer]
You need to have Xdummy working to be able to properly virtualize the DPI.
That means not using Debian or Ubuntu.

Also, 4.4.6 is not a supported version. It has been EOLed.

**Comment by Simba98** [comment]
Hi Totaam,

I suggest the workaround is adding --dpi=96 (or any DPI you want) on BOTH server and client.
Then you do not need DPI to rescale, hence no Xdummy is OK.

Of course, the best is using the newest OS, with Xpra 5.x.

**Reply by totaam** [reply]
Sure, but then you get this DPI for the duration of the session rather than the DPI you should be getting: no handling for different clients, or adding / removing monitors, changing resolution, etc..

**Comment by totaam** [comment]
For the record, please avoid Debian and Ubuntu to get correct DPI and resolution behaviour:
https://github.com/Xpra-org/xpra/issues/3989#issuecomment-1720383100
